paasta_tools/api/views/autoscaler.py

```diff
@@ -30,17 +73,9 @@ def get_autoscaler_count(request):
     instance = request.swagger_data.get("instance")
     cluster = settings.cluster
     soa_dir = settings.soa_dir
-    try:
-        service_config = load_marathon_service_config(
-            service=service,
-            instance=instance,
-            cluster=cluster,
-            soa_dir=soa_dir,
-            load_deployments=False,
-        )
-    except Exception:
-        error_message = f"Unable to load service config for {service}.{instance}"
-        raise ApiFailure(error_message, 404)
+
+    instance_type = get_instance_type(service, instance, cluster, soa_dir)
+    service_config = get_service_config(instance_type, service, instance, cluster, soa_dir)
 
     response_body = {
         "desired_instances": service_config.get_instances(),
```

Comment:
Note that this implementation only supports deployment. Perhaps you can update [it](https://github.com/Yelp/paasta/blob/371da7c1c02bcdabb8317502cd2e9f9aa83b8c9d/paasta_tools/kubernetes_tools.py#L928) to support statefulset ? Or return error if this service uses statefulset. You can tell if it is a stateful set or not like [this](https://github.com/Yelp/paasta/blob/371da7c1c02bcdabb8317502cd2e9f9aa83b8c9d/paasta_tools/kubernetes_tools.py#L1025)

Reply:
added to `set_instances_for_kubernetes_service()`.

Reply:
Ohh, good point, get_autoscaled_instances doesn't handle statefulset. Do you mind fixing this? I could also try to fix it tomorrow.

Reply:
Ah is probably the reason why I had trouble testing on statefulsets, I can add this.